compiler/modules.nim

```diff
@@ -115,9 +116,38 @@ proc wantMainModule*(conf: ConfigRef) =
     fatal(conf, gCmdLineInfo, "command expects a filename")
   conf.projectMainIdx = fileInfoIdx(conf, addFileExt(conf.projectFull, NimExt))
 
+proc resolveModuleOverridePaths(conf: ConfigRef) =
+  ## Resolve the target and patch module paths set with `nimscript.patchModule`.
+  ## This resolution needs performed after all module search paths have been defined.
+  ##
+  ## See Also:
+  ## * `nimscript.patchModule`
+  ## * `options.patchModule`
+  let curDir = getCurrentDir()
+  for (target, override) in conf.unresolvedModuleOverrides.pairs:
+    var
+      resolvedTarget = target
+      resolvedOverride = override
+    if not target.isAbsolute:
+      resolvedTarget = findModule(conf, target, conf.projectFull.string, patch = false).string
+      if resolvedTarget.len == 0:
+        rawMessage(conf, warnCannotOpen, target)
+        continue
+    else:
+      resolvedTarget = target
+    if not override.isAbsolute:
+      resolvedOverride = findModule(conf, override, conf.projectFull.string, patch = false).string
+      if resolvedOverride.len == 0:
+        rawMessage(conf, warnCannotOpen, override)
+    else:
+      resolvedOverride = override
+    if resolvedTarget.len > 0 and resolvedOverride.len > 0:
+      conf.moduleOverridePaths[resolvedTarget] = resolvedOverride
+
 proc compileProject*(graph: ModuleGraph; projectFileIdx = InvalidFileIdx) =
   connectCallbacks(graph)
   let conf = graph.config
+  resolveModuleOverridePaths(conf)
```

Comment:
The target and patch paths need resolved after all search paths are defined.
Is this the best place to do this?